ocaml/alerts/certificate/certificate_check.ml

```diff
@@ -2,10 +2,12 @@ module XenAPI = Client.Client
 module Date = Xapi_stdext_date.Date
 
 let seconds_per_day = 3600. *. 24.
-let seconds_per_30_days = 30. *. seconds_per_day
 
 let days_until_expiry epoch expiry =
-  int_of_float (expiry /. seconds_per_day -. epoch /. seconds_per_day)
+  let days = expiry /. seconds_per_day -. epoch /. seconds_per_day in
+  match Float.sign_bit days with
```

Comment:
why not `days <= 0.`?

Reply:
Because of how floats are [stored](https://en.wikipedia.org/wiki/Single-precision_floating-point_format#IEEE_754_single-precision_binary_floating-point_format:_binary32): every float has a sign bit, this is why `-0.` is an actual value.

Checking the bit should also be the most efficient way to check for this.

Reply:
okay, in this case, i assume you are happy that the certificate has exactly expired, we will report 0 days until expiry, rather than -1?
(since `Float.sign_bit 0. = true`)

Reply:
Yes, I don't think it makes a difference which alert the user is seeing, both have the same priority, and I don't think it's worth tweaking for a case that's _really_ unlikely to happen in a real environment.

Reply:
agreed